--- src/netlib.c.orig      2008-10-21 20:32:05.000000000 -0400
+++ src/netlib.c   2008-10-21 20:35:13.000000000 -0400
@@ -1990,8 +1990,13 @@
 
 #if defined(__CPU_SETSIZE)
 #define NETPERF_CPU_SETSIZE __CPU_SETSIZE
+#if defined(__CPU_SET_S)
+#define NETPERF_CPU_SET(cpu, cpusetp)  __CPU_SET_S(cpu, sizeof (cpu_set_t), cpusetp)
+#define NETPERF_CPU_ZERO(cpusetp)      __CPU_ZERO_S (sizeof (cpu_set_t), cpusetp) 
+#else
 #define NETPERF_CPU_SET(cpu, cpusetp)  __CPU_SET(cpu, cpusetp)
 #define NETPERF_CPU_ZERO(cpusetp)      __CPU_ZERO (cpusetp)
+#endif
   typedef cpu_set_t netperf_cpu_set_t;
 #else
 #define NETPERF_CPU_SETSIZE sizeof(unsigned long)
