--- src/src/execl.c	1999-07-28 14:46:58.000000000 -0700
+++ newsrc/src/execl.c	2008-02-27 09:23:17.372306000 -0800
@@ -88,7 +88,7 @@ char	*argv[];
 		fprintf(stderr, "%lu loops\n", iter);
 		exit(0);
 		}
-	execl(fullpath, fullpath, "0", dur_str, count_str, start_str, 0);
+	execl(fullpath, fullpath, "0", dur_str, count_str, start_str, (char *)0);
 	printf("Exec failed at iteration %lu\n", iter);
 	perror("Reason");
 	exit(1);
