diff -Naur ext3-tools/fsstress.c ext3-tools.new/fsstress.c
--- ext3-tools/fsstress.c	2004-05-17 04:49:53.000000000 -0300
+++ ext3-tools.new/fsstress.c	2007-07-16 14:26:20.000000000 -0300
@@ -487,7 +487,7 @@
 #endif
 	if (cleanup == 0)
 	{
-	  sprintf(cmd,"rm -rf %s",dirname);
+	  sprintf(cmd,"rm -rf %s/*",dirname);
 	  system(cmd);
 	}	
         loopcntr++;
@@ -1453,7 +1453,7 @@
 	if (!get_fname(FT_ANYm, r, &f, NULL, NULL, &v))
 		append_pathname(&f, ".");
 	total = 0;
-	bzero(&cursor, sizeof(cursor));
+	memset(&cursor, 0x00, sizeof(cursor));
 	do {
 		e = attr_list_path(&f, buf, sizeof(buf), ATTR_DONTFOLLOW,
 			&cursor);
@@ -1471,7 +1471,7 @@
 		return;
 	}
 	which = (int)(random() % total);
-	bzero(&cursor, sizeof(cursor));
+	memset(&cursor, 0x00, sizeof(cursor));
 	ent = 0;
 	aname = NULL;
 	do {
