diff -Nubr ffsb-6.0-rc2-orig/examples/mail_server_8k_1.ffsb ffsb-6.0-rc2/examples/mail_server_8k_1.ffsb
--- ffsb-6.0-rc2-orig/examples/mail_server_8k_1.ffsb	1970-01-01 08:00:00.000000000 +0800
+++ ffsb-6.0-rc2/examples/mail_server_8k_1.ffsb	2011-07-27 15:53:51.354350678 +0800
@@ -0,0 +1,64 @@
+# Mail server simulation.
+
+time=300
+alignio=1
+directio=1
+
+[filesystem0]
+	location=/mnt/ffsb1
+	num_files=1000000
+	num_dirs=1000
+
+	# File sizes range from 1kB to 1MB.
+	size_weight 1KB 10
+	size_weight 2KB 15
+	size_weight 4KB 16
+	size_weight 8KB 16
+	size_weight 16KB 15
+	size_weight 32KB 10
+	size_weight 64KB 8
+	size_weight 128KB 4
+	size_weight 256KB 3
+	size_weight 512KB 2
+	size_weight 1MB 1
+[end0]
+
+[threadgroup0]
+	num_threads=1
+
+	readall_weight=4
+	create_fsync_weight=2
+	delete_weight=1
+
+	write_size=8KB
+	write_blocksize=8KB
+
+	read_size=8KB
+	read_blocksize=8KB
+
+	[stats]
+		enable_stats=1
+		enable_range=1
+
+		msec_range    0.00      0.01
+		msec_range    0.01      0.02
+		msec_range    0.02      0.05
+		msec_range    0.05      0.10
+		msec_range    0.10      0.20
+		msec_range    0.20      0.50
+		msec_range    0.50      1.00
+		msec_range    1.00      2.00
+		msec_range    2.00      5.00
+		msec_range    5.00     10.00
+		msec_range   10.00     20.00
+		msec_range   20.00     50.00
+		msec_range   50.00    100.00
+		msec_range  100.00    200.00
+		msec_range  200.00    500.00
+		msec_range  500.00   1000.00
+		msec_range 1000.00   2000.00
+		msec_range 2000.00   5000.00
+		msec_range 5000.00  10000.00
+	[end]
+[end0]
+
