diff -Nubr ffsb-6.0-rc2-orig/examples/sequential_reads_256k_1.ffsb ffsb-6.0-rc2/examples/sequential_reads_256k_1.ffsb
--- ffsb-6.0-rc2-orig/examples/sequential_reads_256k_1.ffsb	1970-01-01 08:00:00.000000000 +0800
+++ ffsb-6.0-rc2/examples/sequential_reads_256k_1.ffsb	2011-07-27 15:53:50.961226350 +0800
@@ -0,0 +1,46 @@
+# Large file sequential reads.
+# 1024 files, 100MB per file.
+
+time=30  # 5 min
+alignio=1
+directio=1
+
+[filesystem0]
+	location=/mnt/ffsb1
+	num_files=1024
+	min_filesize=100MB
+	max_filesize=100MB
+	reuse=1
+[end0]
+
+[threadgroup0]
+	num_threads=1
+	readall_weight=1
+	read_blocksize=256KB
+
+	[stats]
+		enable_stats=1
+		enable_range=1
+
+		msec_range    0.00      0.01
+		msec_range    0.01      0.02
+		msec_range    0.02      0.05
+		msec_range    0.05      0.10
+		msec_range    0.10      0.20
+		msec_range    0.20      0.50
+		msec_range    0.50      1.00
+		msec_range    1.00      2.00
+		msec_range    2.00      5.00
+		msec_range    5.00     10.00
+		msec_range   10.00     20.00
+		msec_range   20.00     50.00
+		msec_range   50.00    100.00
+		msec_range  100.00    200.00
+		msec_range  200.00    500.00
+		msec_range  500.00   1000.00
+		msec_range 1000.00   2000.00
+		msec_range 2000.00   5000.00
+		msec_range 5000.00  10000.00
+	[end]
+[end0]
+
