diff -urN ltp-full-20091231_vanilla/runtest/syscalls ltp-full-20091231/runtest/syscalls
--- ltp-full-20091231_vanilla/runtest/syscalls	2010-01-05 17:16:06.000000000 -0800
+++ ltp-full-20091231/runtest/syscalls	2010-01-05 17:27:50.000000000 -0800
@@ -35,11 +35,11 @@
 
 brk01 brk01
 
-capget01 capget01
-capget02 capget02
+#capget01 capget01
+#capget02 capget02
 
-capset01 capset01
-capset02 capset02
+#capset01 capset01
+#capset02 capset02
 
 cacheflush01 cacheflush01
 
