--- a/src/Makefile	2006-09-23 15:17:18.000000000 -0700
+++ b/src/Makefile	2006-09-23 15:17:43.000000000 -0700
@@ -108,7 +108,7 @@
 DEFS = -DHAVE_CONFIG_H -I. -I$(srcdir) -I..
 CPPFLAGS = 
 LDFLAGS = 
-LIBS = 
+LIBS += 
 reaim_OBJECTS =  driver.$(OBJEXT) ram.$(OBJEXT) rand.$(OBJEXT) \
 times_ms.$(OBJEXT) add.$(OBJEXT) creat_clo.$(OBJEXT) \
 drv_funcs.$(OBJEXT) div.$(OBJEXT) disk_funcs.$(OBJEXT) \
@@ -116,7 +116,7 @@
 disk1.$(OBJEXT) disk_src.$(OBJEXT) fun.$(OBJEXT) funcal.$(OBJEXT)
 reaim_DEPENDENCIES =  $(top_builddir)/lib/libltp.a
 reaim_LDFLAGS = 
-CFLAGS = -W -Wall -lm -ffloat-store -g -O -D_GNU_SOURCE -DSHARED_OFILE
+CFLAGS += -W -Wall -lm -ffloat-store -g -O -D_GNU_SOURCE -DSHARED_OFILE
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 LTCOMPILE = $(LIBTOOL) --mode=compile $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
