--- lmbench3.old/src/Makefile	2005-08-22 17:19:54.000000000 -0700
+++ lmbench3.new/src/Makefile	2006-04-22 13:10:59.000000000 -0700
@@ -165,41 +165,6 @@ debug:
 assembler:
 	@env CFLAGS=-O MAKE="$(MAKE)" MAKEFLAGS="$(MAKEFLAGS)" CC="${CC}" OS="${OS}" ../scripts/build asm
 
-bk.ver: ../SCCS/s.ChangeSet
-	rm -f bk.ver
-	-echo `bk prs -hr+ -d'$$if(:SYMBOL:){:SYMBOL: }:UTC:' ../ChangeSet;` > bk.ver
-	touch bk.ver
-
-dist: bk.ver
-	@if [ "X`cd ..; bk sfiles -c`" != "X" ]; then \
-		echo "modified files!"; \
-		false; \
-	 fi
-	@if [ "X`cd ..; bk pending`" != "X" ]; then \
-		echo "pending changes!"; \
-		false; \
-	 fi
-	cd ..; \
-		SRCDIR=`pwd`; \
-		DIR=`basename $${SRCDIR}`; \
-		VERSION=`cat src/bk.ver| awk '{print $$1;}' | sed -e 's/Version-//g'`; \
-		cd ..; \
-		bk clone $${DIR} /tmp/lmbench-$${VERSION}; \
-		cd /tmp/lmbench-$${VERSION}; \
-		bk sfiles | xargs touch; \
-		sleep 5; \
-		bk get -s; \
-		for d in doc results scripts src; do \
-			cd $$d; bk get -s; cd ..; \
-		done; \
-		bk sfiles -U -g | xargs touch; \
-		cd src; \
-		make bk.ver; \
-		cd /tmp; \
-		tar czf $${SRCDIR}/../lmbench-$${VERSION}.tgz \
-			lmbench-$${VERSION}; \
-		rm -rf /tmp/lmbench-$${VERSION};
-
 get $(SRCS):
 	-get -s $(SRCS)
 
@@ -228,9 +193,9 @@ testmake: $(SRCS) $(UTILS) # used by scr
 	install install-target dist get edit get-e clean clobber \
 	share depend testmake
 
-$O/lmbench : ../scripts/lmbench bk.ver
+$O/lmbench : ../scripts/lmbench 
 	rm -f $O/lmbench
-	sed -e "s/<version>/`cat bk.ver`/g" < ../scripts/lmbench > $O/lmbench
+	sed -e "s/<version>/666/g" < ../scripts/lmbench > $O/lmbench
 	chmod +x $O/lmbench
 
 $O/lmbench.a: $(LIBOBJS)
--- src/scripts/config-run.old	2006-06-07 12:46:15.000000000 -0700
+++ src/scripts/config-run	2006-06-07 12:52:07.000000000 -0700
@@ -115,9 +115,12 @@
 three benchmark processes reading data and doing the measurements.
 
 EOF
-	echo $ECHON "Job placement selection: $ECHOC"
+	echo $ECHON "Job placement selection [DEFAULT: 1]: $ECHOC"
 	read LMBENCH_SCHED
 	AGAIN=N
+	if [ "$LMBENCH_SCHED" == "" ]; then
+		LMBENCH_SCHED=1
+	fi
 	case "$LMBENCH_SCHED" in
 	    1) LMBENCH_SCHED=DEFAULT;;
 	    2) LMBENCH_SCHED=BALANCED;;
@@ -657,13 +660,13 @@
 
 EOF
 
-echo $ECHON "Mail results [default yes] $ECHOC"
+echo $ECHON "Mail results [default no] $ECHOC"
 read MAIL
 case $MAIL in 
-    [Nn]*)	MAIL=no
-		echo OK, no results mailed.
+    [Yy]*)	MAIL=yes
+		echo OK, results will be mailed.
 		;;
-    *)		MAIL=yes
+    *)		MAIL=no
 		;;
 esac
 
